float32 pos_x
float32 pos_y
float32 orientation_x
float32 orientation_y